src/DiagramGenerator/Diagram/Board.php

```diff
@@ -380,11 +380,12 @@ protected function getMaxPieceHeight()
      */
     protected function cacheImage($remoteImageUrl, $cachedFilePath)
     {
+        $cachedFilePathTmp = $cachedFilePath . rand(1000000,9999999);
```

Comment:
as far as performance is concerned, `rand()` is probably the worse of all the it's brothers. Probably the thing to be used here is [uniqid()](http://php.net/manual/en/function.uniqid.php)